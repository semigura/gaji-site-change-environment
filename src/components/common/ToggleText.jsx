import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import { SlideDown } from 'react-slidedown';
import 'react-slidedown/lib/slidedown.css';

import css from './ToggleText.module.scss';

class ToggleText extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      isSP: false,
    };
    this.handleToggle = this.handleToggle.bind(this);
    this.checkMedia = this.checkMedia.bind(this);
  }

  checkMedia = () => {
    if (window.matchMedia('(max-width: 939px)').matches) {
      this.setState({ isSP: true });
    } else {
      this.setState({ isSP: false });
    }
  };

  componentDidMount = () => {
    window.addEventListener('resize', this.checkMedia);
    this.checkMedia();
  };

  componentWillUnmount = () => {
    window.removeEventListener('resize', this.checkMedia);
  };

  handleToggle = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  };

  render() {
    const { isOpen, isSP } = this.state;
    const { className, children } = this.props;
    const classes = classNames(css.ToggleText, className);
    const buttonClasses = classNames(css.ToggleText__Button, {
      [css.isOpen]: isOpen,
    });
    return (
      <div className={classes}>
        {isSP ? (
          <>
            <SlideDown>{isOpen && children}</SlideDown>
            <button
              type="button"
              className={buttonClasses}
              onClick={this.handleToggle}
              aria-label="開く"
            />
          </>
        ) : (
          children
        )}
      </div>
    );
  }
}

ToggleText.propTypes = {
  className: PropTypes.string,
  children: PropTypes.element.isRequired,
};

ToggleText.defaultProps = {
  className: undefined,
};

export default ToggleText;
