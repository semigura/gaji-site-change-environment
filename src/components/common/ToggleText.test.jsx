import { shallow } from 'enzyme';
import React from 'react';

import ToggleText from './ToggleText';

describe('<ToggleText />', () => {
  it('renders correctly when PC', () => {
    window.matchMedia = jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
    }));

    const wrapper = shallow(
      <ToggleText>
        <p className="Index__LeadText">
          Gaji-Laboは変化や刺激を必要としているチームの「変数」となり、チームの殻をやわらかくする会社です。チームが持っているそれぞれの物差しに合わせ、生まれる価値や熱量を見つけたり膨らませたり整えたりすること。そのために影で伴走する黒子の役割が、私たちGaji-Laboの仕事です。
        </p>
      </ToggleText>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when SP', () => {
    window.matchMedia = jest.fn().mockImplementation(query => ({
      matches: true,
      media: query,
    }));

    const wrapper = shallow(
      <ToggleText>
        <p className="Index__LeadText">
          Gaji-Laboは変化や刺激を必要としているチームの「変数」となり、チームの殻をやわらかくする会社です。チームが持っているそれぞれの物差しに合わせ、生まれる価値や熱量を見つけたり膨らませたり整えたりすること。そのために影で伴走する黒子の役割が、私たちGaji-Laboの仕事です。
        </p>
      </ToggleText>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleToggle', () => {
    window.matchMedia = jest.fn().mockImplementation(query => ({
      matches: true,
      media: query,
    }));

    const wrapper = shallow(
      <ToggleText>
        <p className="Index__LeadText">
          Gaji-Laboは変化や刺激を必要としているチームの「変数」となり、チームの殻をやわらかくする会社です。チームが持っているそれぞれの物差しに合わせ、生まれる価値や熱量を見つけたり膨らませたり整えたりすること。そのために影で伴走する黒子の役割が、私たちGaji-Laboの仕事です。
        </p>
      </ToggleText>,
    );

    it('renders correctly', () => {
      wrapper.find('.ToggleText__Button').simulate('click');
      expect(
        wrapper.find('.ToggleText__Button').hasClass('isOpen'),
      ).toBeTruthy();
    });
  });
});
