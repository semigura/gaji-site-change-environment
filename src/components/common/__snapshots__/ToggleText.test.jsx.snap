// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleText /> renders correctly when PC 1`] = `
<div
  className="ToggleText"
>
  <p
    className="Index__LeadText"
  >
    Gaji-Laboは変化や刺激を必要としているチームの「変数」となり、チームの殻をやわらかくする会社です。チームが持っているそれぞれの物差しに合わせ、生まれる価値や熱量を見つけたり膨らませたり整えたりすること。そのために影で伴走する黒子の役割が、私たちGaji-Laboの仕事です。
  </p>
</div>
`;

exports[`<ToggleText /> renders correctly when SP 1`] = `
<div
  className="ToggleText"
>
  <ForwardRef />
  <button
    aria-label="開く"
    className="ToggleText__Button"
    onClick={[Function]}
    type="button"
  />
</div>
`;
